lib/socket_proxy/application.ex

```diff
@@ -5,21 +5,42 @@ defmodule SocketProxy.Application do
   require Logger
 
   def start(_type, _args) do
-    Logger.info("Starting SocketProxy.Application")
+    if Mix.env == :test do
+      Supervisor.start_link([], strategy: :one_for_all, name: SocketProxy.Supervisor)
+    else
+      Logger.info("Starting SocketProxy.Application")
 
-    listen_port = Application.get_env(:socket_proxy, :listen_port) || 4000
-    destinations = Application.get_env(:socket_proxy, :destinations) || []
+      listen_port = Application.get_env(:socket_proxy, :listen_port) || get_listen_port()
```

Comment:
Is `:socket_proxy, :listen_port` ever put in the environment anymore?

Reply:
Test suite. And I was thinking potentially if you wanted to start it from within an `iex` session or something. I don't love changing code to accommodate tests, but I was thinking of it along the lines of how we do DI to allow testability.